### DISCUSSION TITLE
Should total per week?

### DISCUSSION THREAD

**Discussion by miquecg** [jotaen/klog#212] [Help / Question] [answered]
Hi,

I've seen klog has the handy feature of expected total hours for a given day but I'm wondering if that could also work for a week total. So, if I add a `should-total` per day record, could I create a report at any moment that shows the balance for that week considering the target?

For my needs it's also important that I don't mix different projects while logging work. I guess having different files would be enough. In that case, can reports summarize several files together?

Thanks for this great tool!

**Comment by jotaen** [comment, marked as answer]
The should total works on day-level. Maybe it works for you if you specify the should total at the first day of the week, like:

```
2022-09-12 (40h!)
This is Monday
  8h Work

2022-09-13
This is Tuesday
  8h Work
```

Technically speaking, the entire `40h!` refers to Monday. However, if you evaluate the records of the week, it sums up all values from the matching records. E.g.: `klog total --period 2022-W37 --diff myfile.klg` (i.e. total time of week 37) would yield:

```
Total: 16h
Should: 40h!
Diff: -24h
(In 2 records)
```

Regarding your second question, all evaluation commands (`klog total`, `klog report`, `klog tags`, `klog today`) accept multiple files or [bookmarks](https://klog.jotaen.net/#bookmarks) as input, e.g. `klog total one.klg two.klg three.klg @four @five`

**Reply by miquecg** [reply]
Hey, it's a valid workaround 😄 

Following your example, would the flag `--now` still work as per the docs?

> When using `klog today --diff --now`, it also forecasts the end-time at which your time goal will be reached.

It is simply more convenient to just know when you'd reach the target instead of doing the calculations yourself.

**Reply by jotaen** [reply]
The `--now` flag always assumes any open time range as closed “right now”. So let’s say it’s 18:36 right now, then `14:00 - ?` will be (transiently) evaluated as `14:00 - 18:36` → `4h36m`. This only works for open ranges at today’s or yesterday’s date, however. (klog will display a warning otherwise.) So as long as this is the case, `klog today --diff --now` works as usual.

Just for further context of how klog works under the hood: if you do something like `klog today --now --period=2022-W34 --tag=work one.klg @two`, then klog does the following:

1. Read all inputs. (In this case the data from both the file `one.klg` and the bookmarked file `@two`.)
2. Apply filters. In this case: only consider records/entries within the period `2022-W34` **and** whose summary contains the tag `#work`.
3. Assume any open time ranges as closed “right now”
4. Evaluate the resulting data

This data flow is always the same, so klog doesn’t make any assumptions about how you structure the data in your files.

In order to get more familiar with how klog works, you could also create a test file to play around with the filters and functionality. Apart from the `--now` flag, klog’s evaluations are deterministic, so it always outputs the same result for the same input data.

**Reply by miquecg** [reply]
Thanks! I'll give it a try.
